### PR TITLE
[FEATURE] view::persist

### PR DESCRIPTION
**Pull request by h-2**
necessary for file I/O snippets and tutorials

### CODE REVIEW

File: test/snippet/range/view/persist.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+//! [usage]
+// this doesn't work:
+// auto v = "ACGT"_dna4 | view::to_char;
+
+// this does, but is less pretty:
+auto vec = "ACGT"_dna4;
+auto v = vec | view::to_char;
+
+// using view::persist you can bind the temporary directly:
+auto v2 = "ACGT"_dna4 | view::persist | view::to_char;
```
[the remark this answers is not in the document]
Comment (h-2) [reply test/snippet/range/view/persist.cpp]: I will add both to the example!

File: include/seqan3/range/view/persist.hpp [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+
+namespace seqan3::detail
+{
+
+// ============================================================================
+//  view_persist
+// ============================================================================
+
+/*!\brief The type returned by seqan3::view::persist.
+ * \tparam urng_t    The type of the underlying ranges, must satisfy std::ranges::InputRange.
```
Comment (smehringer) [inline include/seqan3/range/view/persist.hpp]: range**~s~**
~satisfy~ model
and remove the extra space before urange_t

File: include/seqan3/range/view/persist.hpp [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+     */
+    view_persist() = default;
+    constexpr view_persist(view_persist const & rhs) = default;
+    constexpr view_persist(view_persist && rhs) = default;
+    constexpr view_persist & operator=(view_persist const & rhs) = default;
+    constexpr view_persist & operator=(view_persist && rhs) = default;
+    ~view_persist() = default;
+
+    /*!\brief Construct from another range.
+     * \param[in] _urange The underlying range.
```
Comment (smehringer) [inline include/seqan3/range/view/persist.hpp]: tparam missing

File: test/snippet/range/view/persist.cpp [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/range/view/persist.hpp>
+#include <seqan3/range/view/to_char.hpp>
+
+using namespace seqan3;
+using namespace seqan3::literal;
+
+int main()
+{
+//! [usage]
```
Comment (smehringer) [inline test/snippet/range/view/persist.cpp]: include the `main` and the `#includes` into the example. I think larger examples should be copy and pastable.

File: include/seqan3/range/view/persist.hpp (hunk 3 above)
Comment (h-2) [reply include/seqan3/range/view/persist.hpp]: it doesnt have a template parameter so I don't need to document it :laughing:

File: test/snippet/range/view/persist.cpp (hunk 4 above)
Comment (h-2) [reply test/snippet/range/view/persist.cpp]: lets discuss this with the other snippets

File: test/unit/range/view/view_persist_test.cpp [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+//  test templates
+// ============================================================================
+
+TEST(view_persist, delegate_to_view_all)
+{
+    std::string vec{"foo"};
+
+    // pipe notation
+    auto v = vec | view::persist;
+    EXPECT_EQ("foo", std::string(v));
```
Comment (smehringer) [inline test/unit/range/view/view_persist_test.cpp]: `{ }` for construction ?

**Comment by h-2** [comment]
@rrahn can we merge this? all has passed!

**Pull request merged**